toolchain/check/handle_binding_pattern.cpp

```diff
@@ -185,13 +197,22 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
     }
 
     case Parse::NodeKind::LetIntroducer: {
-      cast_type_id = context.AsCompleteType(cast_type_id, [&] {
-        CARBON_DIAGNOSTIC(IncompleteTypeInLetDecl, Error,
-                          "`let` binding has incomplete type `{0}`",
-                          SemIR::TypeId);
-        return context.emitter().Build(type_node, IncompleteTypeInLetDecl,
-                                       cast_type_id);
-      });
+      cast_type_id = context.AsCompleteType(
+          cast_type_id,
+          [&] {
+            CARBON_DIAGNOSTIC(IncompleteTypeInLetDecl, Error,
+                              "`let` binding has incomplete type `{0}`",
+                              SemIR::TypeId);
+            return context.emitter().Build(type_node, IncompleteTypeInLetDecl,
+                                           cast_type_id);
+          },
+          [&] {
+            CARBON_DIAGNOSTIC(AbstractTypeInLetDecl, Error,
+                              "`let` binding has abstract type `{0}`",
+                              SemIR::TypeId);
+            return context.emitter().Build(type_node, AbstractTypeInLetDecl,
+                                           cast_type_id);
+          });
```

Comment:
I think this case should be permitted, as a `let` binding is forming a value not an object.